samcli/lib/list/resources/resource_mapping_producer.py

```diff
@@ -132,11 +132,11 @@ def produce(self):
             response = self.get_resources_info()
             for deployed_resource in response["StackResources"]:
                 resource_data = ResourcesDef(
-                    LogicalResourceId=deployed_resource["LogicalResourceId"],
-                    PhysicalResourceId=deployed_resource["PhysicalResourceId"],
+                    LogicalResourceId=deployed_resource.get("LogicalResourceId"),
+                    PhysicalResourceId=deployed_resource.get("PhysicalResourceId"),
                 )
                 resources_list.append(dataclasses.asdict(resource_data))
-                seen_resources.add(deployed_resource["LogicalResourceId"])
+                seen_resources.add(deployed_resource.get("LogicalResourceId"))
```

Comment:
If `deployed_resource.get("LogicalResourceId")` returns None, None gets added to the seen_resources set. Should we avoid adding None here and check it with an if condition instead?

Reply:
We could to be extra safe, but in this case if a resource is already deployed it must have a logical ID and a Physical ID, so we should be okay.

Reply:
Right makes sense, in that case should we keep the [] brackets for getting the LogicalResourceId and PhysicalResourceId if these keys are always assumed to be present?

Reply:
I do think it's still better practice to use `.get()` as it's a safer operation here. If downstream there's an issue we can deal with it there, but at least we won't be getting a `KeyError` here.

Reply:
Fair enough